ocaml/xcp-rrdd/bin/rrdd/rrdd_monitor.ml

```diff
@@ -51,12 +51,13 @@ let merge_new_dss rrdi dss =
     !Rrdd_shared.enable_all_dss || ds.ds_default
   in
   let default_dss = StringMap.filter should_enable_ds dss in
+  let ds_names = Rrd.ds_names rrdi.rrd in
```

Comment:
How long is this list? If it is not short, the test below is expensive. An alternative could be to create a set, which has O(log n) complexity for membership or even a hash table.